packages/docusaurus-theme-common/src/index.ts

```diff
@@ -5,161 +5,49 @@
-  type MultiColumnFooter,
-  type SimpleFooter,
-  type Footer,
-  type FooterLogo,
-  type FooterLinkItem,
-  type ColorModeConfig,
+export {useThemeConfig} from './utils/useThemeConfig';
+export type {
+  ThemeConfig,
+  UserThemeConfig,
+  MultiColumnFooter,
+  SimpleFooter,
+  Footer,
+  FooterLogo,
+  FooterLinkItem,
+  ColorModeConfig,
 } from './utils/useThemeConfig';
-export {
-  DocSidebarItemsExpandedStateProvider,
-  useDocSidebarItemsExpandedState,
-} from './contexts/docSidebarItemsExpandedState';
-export {DocsVersionProvider, useDocsVersion} from './contexts/docsVersion';
-export {DocsSidebarProvider, useDocsSidebar} from './contexts/docsSidebar';
-export {DocProvider, useDoc} from './contexts/doc';
-export type {DocContextValue} from './contexts/doc';
-
-export {createStorageSlot, listStorageKeys} from './utils/storageUtils';
-
-export {useAlternatePageUtils} from './utils/useAlternatePageUtils';
-
-export {
-  parseCodeBlockTitle,
-  parseLanguage,
-  parseLines,
-  containsLineNumbers,
-  getPrismCssVariables,
-} from './utils/codeBlockUtils';
 
-export {
-  docVersionSearchTag,
-  DEFAULT_SEARCH_TAG,
-  useContextualSearchFilters,
-} from './utils/searchUtils';
+export {useColorMode} from './contexts/colorMode';
 
-export {
-  isDocsPluginEnabled,
-  useDocById,
-  findSidebarCategory,
-  findFirstCategoryLink,
-  useCurrentSidebarCategory,
-  isActiveSidebarItem,
-  useSidebarBreadcrumbs,
-  useDocsVersionCandidates,
-  useLayoutDoc,
-  useLayoutDocsSidebar,
-  useDocRouteMetadata,
-} from './utils/docsUtils';
+export {usePrismTheme} from './hooks/usePrismTheme';
+export {getPrismCssVariables} from './utils/codeBlockUtils';
 
-export {useTitleFormatter} from './utils/generalUtils';
+export {useCurrentSidebarCategory} from './utils/docsUtils';
 
 export {usePluralForm} from './utils/usePluralForm';
 
-export {useLocationChange} from './utils/useLocationChange';
+export {PageMetadata, HtmlClassNameProvider} from './utils/metadataUtils';
 
-export {useCollapsible, Collapsible} from './components/Collapsible';
+export {ThemeClassNames} from './utils/ThemeClassNames';
 
-export {
-  useDocsPreferredVersion,
-  useDocsPreferredVersionByPluginId,
-  DocsPreferredVersionContextProvider,
-} from './contexts/docsPreferredVersion';
+export {useContextualSearchFilters} from './utils/searchUtils';
 
-export {duplicates, uniq} from './utils/jsUtils';
+// TODO Should these APIs bellow be public?
 
-export {ThemeClassNames} from './utils/ThemeClassNames';
+export {useBackToTopButton} from './hooks/useBackToTopButton';
+export {useWindowSize} from './hooks/useWindowSize';
 
 export {
-  AnnouncementBarProvider,
-  useAnnouncementBar,
-} from './contexts/announcementBar';
-
-export {useLocalPathname} from './utils/useLocalPathname';
+  NavbarSecondaryMenuFiller,
+  type NavbarSecondaryMenuComponent,
+} from './contexts/navbarSecondaryMenu/content';
 
 export {
   translateTagsPageTitle,
   listTagsByLetters,
   type TagLetterEntry,
```

Comment:
These don't need to be public IMO

Reply:
I'll keep those public because these are quite stable, and the tags pages/components (docs, blog) look clean enough to mark them as safe to swizzle. Ultimately we'll always end up with some niche public apis like that if we want no internal components in swizzle